### PR TITLE
Add ASCII-only option, to mimic default RE2 behavior

### PR DESCRIPTION
**Pull request by bzz**
This is a workaround, motivated by the difference in handling non-valid UTF8
bytes that Oniriguma has, compared to Go's default RE2.

See https://github.com/src-d/enry/issues/225#issuecomment-490043281

Summary of changes:
 - use modern [version of Oniguruma](https://github.com/kkos/oniguruma/releases) as it's well-maintained and is constantly improved
 - ci: test on latest 2 Go versions
 - c: `NewOnigRegex()` calls to `onig_initialize()` [1]
 - c: prevent `NewOnigRegex()` from hard-coding UTF8
 - go: expose new `MustCompileASCII()` \w default character class matching only ASCII
 - go: `MustCompile()` refactored, `initRegexp()` extracted for common UTF8/ASCII logic

Encoding was not exposed on Go API level intentionally for simplicity,
in order to avoid introducing complex struct type [2] to API surface.

 1. https://github.com/kkos/oniguruma/blob/83572e983928243d741f61ac290fc057d69fefc3/doc/API#L6
 2. https://github.com/kkos/oniguruma/blob/83572e983928243d741f61ac290fc057d69fefc3/src/oniguruma.h#L121

Signed-off-by: Alexander Bezzubov <bzz@apache.org>

### CODE REVIEW

**Comment by bzz** [comment]
CI is great and it's ready to be merged. 

As I do not have rights neither to push to this repo, nor add reviews - \cc @kuba-- @creachadair  for a quick review.

File: .travis.yml
Comment (creachadair) [inline .travis.yml]:
```suggestion
  - sudo apt-get install -y dpkg  # dpkg >= 1.17.5ubuntu5.8, which fixes https://bugs.launchpad.net/ubuntu/+source/dpkg/+bug/1730627
  - sudo dpkg -i libonig-dev_6.9.1-1_amd64.deb
```

File: regex.go
Comment (creachadair) [inline regex.go]:
```suggestion
// MustCompileASCII is equivalent to MustCompile, but with the encoding restricted to ASCII.
```

**Review by erizocosmico — APPROVED** [review]

File: regex.go
Comment (creachadair) [inline regex.go]: It would be better not to use the named returns here:

```go
return initRegexp(&Regexp{pattern: pattern, encoding: C.ONIG_ENCODING_UTF8}, option)
```

The names are pre-existing but I also suggest dropping them, since they don't add any documentary value.

File: regex.go
Comment (creachadair) [inline regex.go]: As 51, _mutatis mutandis_.

File: regex.go
Comment (creachadair) [inline regex.go]: While we're touching it anyway, I recommend we clean up the naming style: `errorCode` here and below.

File: .travis.yml
Comment (kuba--) [reply .travis.yml]: We install `libonig-dev` via `apt addon`. If we really want to install this particular version maybe it's possible specify `libonig-dev_6.9.1-1`, otherwise I would remove `libonig-dev` from addons.

File: .travis.yml
Comment (kuba--) [inline .travis.yml]: why do we need _non dev_ version?

**Comment by bzz** [comment]
Thank you for the prompt reviews! Feedback was addressed in last 2 commits and it's ready for another round.

File: regex.go
Comment (bzz) [reply regex.go]: addressed in e928d1f

File: regex.go
Comment (bzz) [reply regex.go]: thank you! addressed in e928d1f

File: regex.go
Comment (creachadair) [inline regex.go]:
```suggestion
// NewRegexp creates and initializes a new Regexp with the given pattern and option.
```

**Review by creachadair — APPROVED** [review]
One more minor comment fixing a typo, otherwise looks good to me!

File: chelper.c
Comment (kuba--) [inline chelper.c]: I'm ok with this (onig_initialize is kind of once function), but maybe in this case it's easier to call:
`extern int onig_initialize_encoding(OnigEncoding enc)`

File: .travis.yml
Comment (kuba--) [inline .travis.yml]: build fails > `{$ONIGURUMA_VERSION}}`

File: .travis.yml
Comment (bzz) [reply .travis.yml]: because it's a direct dependency of the dev version

File: .travis.yml
Comment (kuba--) [inline .travis.yml]: s/ancent/ancient/

**Comment by bzz** [comment]
@kuba-- @creachadair thank you for kind feedback! All has been addressed and CI is green :shipit: 

Waiting for this to be merged&released to go forward with https://github.com/src-d/enry/pull/227 that is the last thing blocking enry/v2 release with go modules.

@smola do you know who has right GH permissions to merge/cut a release of go-oniguruma? `MAINTAINER` files seems to be missing here.

File: .travis.yml
Comment (bzz) [reply .travis.yml]:
```suggestions
before_install: # install oniguruma manually as trusty has only ancient 5.x
```

oh, suggestions do not work in reply

Comment (bzz) [reply .travis.yml]: addressed in d1a3c51

**Review by kuba-- — APPROVED** [review]

**Comment by bzz** [comment]
@kuba-- thank you for merging! Would you be so kind to ping me, when the next (I guess `v1.1.0`) version that includes this changes is released? Thanks!

**Comment by kuba--** [comment]
@bzz - pushed v1.1.0

**Pull request merged**
